File: javaBack/ResourceServer/src/main/java/com/notes/models/SyntaxType.java
package com.notes.models;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Getter;

@Getter
public enum SyntaxType {
   @JsonProperty("PLAINTEXT")
   PLAINTEXT(0, "PLAINTEXT"),
   @JsonProperty("MARKDOWN")
   MARKDOWN(1, "MARKDOWN");
   final int id;
   final String name;

   SyntaxType(int id, String name) {
      this.id = id;
      this.name = name;
   }

   public static SyntaxType valueOf(int i) {
      return switch (i) {
         case 0 -> PLAINTEXT;
         case 1 -> MARKDOWN;
         default -> throw new IllegalArgumentException("Unknown syntax type: " + i);
      };
   }
}

File: Backend/csharp/NoteS/models/entity/SyntaxType.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Diagnostics.CodeAnalysis;
using System.Text.Json.Serialization;

namespace NoteS.models.entity;

[Table("custom_types_syntax")]
public class SyntaxType
{
    [Column("id")] [Key] public int Num { get; init; }
    [Column("syntax")] public SyntaxTypeName Name { get; init; }
    public static readonly SyntaxType Plain = new(0, SyntaxTypeName.PLAINTEXT);
    public static readonly SyntaxType Markdown = new(1, SyntaxTypeName.MARKDOWN);

    [JsonConverter(typeof(JsonStringEnumConverter))]
    public enum SyntaxTypeName
    {
        PLAINTEXT,
        MARKDOWN
    }

    public static int? TypeToNum(SyntaxType? syntaxType)
    {
        return syntaxType?.Num;
    }
    [return: NotNullIfNotNull(nameof(syntaxType))]
    public static SyntaxType? NameToType(SyntaxTypeName? syntaxType)
    {
        if (syntaxType == null) return null;
        switch (syntaxType)
        {
            case SyntaxTypeName.PLAINTEXT: return Plain;
            case SyntaxTypeName.MARKDOWN: return Markdown;
        }

        throw new ArgumentException();
    }

    [return: NotNullIfNotNull(nameof(num))]
    public static SyntaxType? NumToType(int? num)
    {
        if (num == null) return null;
        switch (num)
        {
            case 0: return Plain;
            case 1: return Markdown;
        }

        throw new ArgumentException(); //TODO: другая ошибка
    }

    protected SyntaxType()
    {
    }

    protected SyntaxType(int num, SyntaxTypeName name)
    {
        Num = num;
        Name = name;
    }
}